### ISSUE TITLE
Unable to return unescaped XML result

### ISSUE THREAD

**Issue by dcolakchile** [DigDes/SoapCore#371]
The library is encoding return XML result, so instead of getting proper XML in the result client gets HTML encoded content.

<?xml version="1.0" encoding="utf-8"?><s:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"><s:Body><getStatusReportResponse xmlns="http://tempuri.org/"><getStatusReportResult>&lt;executions /&gt;</getStatusReportResult></getStatusReportResponse></s:Body></s:Envelope>

How to return string with XML response that is NOT escaped?

Thank you.

**Comment by dcolakchile** [comment]
Is this project alive?

**Comment by kotovaleksandr** [comment]
Yes, alive. For minimal you can provide sample of service with model. And we open for PR, code is not complex.

**Comment by mstavrev** [comment]
I have the same issue with custom soap clients that do not properly support escape sequences on the text elements of the reaponse.

Basically any model that contains a string propery produces such output, giving that its actual value contains a double or single quote character when returned by a service instance.

**Comment by mstavrev** [comment]
@kotovaleksandr, can you hint me where to make such a change, I can create a PR about it?

**Comment by kotovaleksandr** [comment]
@mstavrev Serialization (and escaping) make .NET DataContract(Xml)Serializer.
How client can parse message with unescaped xml? Its invalid message.
I don't understand how it should work:)

**Comment by mstavrev** [comment]
I thought the same prior from moving from a legacy .NET .asmx project to soapcore. It turned out that the classic WS .asmx produced text elements containing stringa with quote character that wa not escaped. It seems the double and single quote characters are not escaped in this context (text within xml element, not an attribute) as it does not break the XML structure.

**Comment by kotovaleksandr** [comment]
@mstavrev Try disable in SoapMessageEncoder CheckCharacters property (in all XmlWriterSettings references), like this:
![image](https://user-images.githubusercontent.com/2936914/81163552-9f14de00-8f97-11ea-94e0-7f5017737f12.png)

If it help, it will be possible to add a setting so that the library user chooses the best behavior for himself

**Comment by mstavrev** [comment]
Thanks, I will look at it later today. Meanwhile, I tried to find documentation refernce of what and how needs be esacaped in general XML. According to this paragraph of the standard https://www.w3.org/TR/xml/#syntax  

> [Definition: All text that is not markup constitutes the character data of the document.] 

> The ampersand character (&) and the left angle bracket (<) MUST NOT appear in their literal form, except when used as markup delimiters, or within a comment, a processing instruction, or a CDATA section. If they are needed elsewhere, they MUST be escaped using either numeric character references or the strings " &amp; " and " &lt; " respectively.

If you read on, basically, even the right angle braket does not need be escaped, unless it appears on markup level where it constitutes different than its pure literal meaning.

**Comment by mstavrev** [comment]
> 
> 
> @mstavrev Try disable in SoapMessageEncoder CheckCharacters property (in all XmlWriterSettings references), like this:
> ![image](https://user-images.githubusercontent.com/2936914/81163552-9f14de00-8f97-11ea-94e0-7f5017737f12.png)
> 
> If it help, it will be possible to add a setting so that the library user chooses the best behavior for himself

Unfortunately, this does not change anything on the output. I debugged the custom build just to be sure the setting had been added. Any other idea?

**Comment by kotovaleksandr** [comment]
@mstavrev I think is not possible in modern .NET:
https://stackoverflow.com/questions/27775793/is-closing-tag-a-valid-character-in-xml

But, i found code which responsible for string serialization:
https://github.com/DigDes/SoapCore/blob/b656fa433cc74d57ad7fc8478c6137e59437ec9f/src/SoapCore/ServiceBodyWriter.cs#L286

If you want, you can add new setting for disable escaping and write to xml raw string (its very dangerous). Please send PR with test.

**Comment by oleg-varlamov** [comment]
@kotovaleksandr Same problem. In a new project, it was necessary to write raw xml as CDATA in the tag. For a long time I could not figure out what was wrong with my injector, which makes some changes in SOAP Message. Now I realized that the matter is in the library. It would be very nice if developers added a setting that allows not to escape content.

I will most likely make a custom build of SoapCore, as this behavior is needed right now.

**Comment by kotovaleksandr** [comment]
@oleg-varlamov I don't know how serialize string through XmlSerializer without escaping.
You can add wrapper properties to model which unescape getters, like this:
https://stackoverflow.com/questions/1379888/how-do-you-serialize-a-string-as-cdata-using-xmlserializer
